application/bsdd_utils.py

```diff
@@ -0,0 +1,57 @@
+import database
+import json
+
+def get_hierarchical_bsdd(id):
+    with database.Session() as session:
+        model = session.query(database.model).filter(
+            database.model.code == id).all()[0]
+
+        hierarchical_bsdd_results = {}
+        if model.status_bsdd != 'n':
+            for bsdd_result in model.tasks[2].results:
+                
+                bsdd_result = bsdd_result.serialize()
+             
+                if bsdd_result["instance_id"]:
+                    bsdd_result['global_id'] = session.query(database.ifc_instance).filter(database.ifc_instance.id == bsdd_result["instance_id"]).all()[0].global_id
```

Comment:
No reason to do twice the same query.

```
inst = session.query(database.ifc_instance).filter(database.ifc_instance.id == bsdd_result["instance_id"]).all()[0]
bsdd_result['global_id'], bsdd_result['ifc_type'] = inst.global_id, inst.ifc_type
```

Reply:
And also cache the results as there might be multiple results relying on the same instance.